DataFormats/Scouting/src/classes_def.xml

```diff
@@ -41,11 +41,42 @@
   <class name="Run3ScoutingVertex" ClassVersion="3">
       <version ClassVersion="3" checksum="316446070"/>
   </class>
-  <class name="Run3ScoutingElectron" ClassVersion="5">
+  <class name="Run3ScoutingElectron" ClassVersion="6">
       <version ClassVersion="3" checksum="1086011373"/>
       <version ClassVersion="4" checksum="1250202632"/>
       <version ClassVersion="5" checksum="2230390721"/>
-  </class>
+      <version ClassVersion="6" checksum="2815634332"/>
+  </class>
+
+  <!-- Adding ioread rules for backwards compatibility -->
+  
+  <ioread sourceClass="Run3ScoutingElectron" version="[3-5]" 
+	  source="float d0_;" 
+	  targetClass="Run3ScoutingElectron" target="trkd0_">
+    <![CDATA[
+	     trkd0_.clear();
```

Comment:
I'm confused why this `clear()` is needed (but I acknowledge the evicende). @pcanal Could you comment if from ROOT point of view this `clear()` serves a purpose, or if the need for one is a sign of something else going on?

Reply:
Since `trkd0` is a 'new' data member, when reading back an old version on of the object, this data member will be untouched.  So indeed the `clear` is necessary to insure that the vector contains only new data (albeit usually `CMSSW` does not "reuse" objects so it should be a no-op in practice).  Note that if more than one rule want to put elements in the vector, the `clear` should (of course) be called only in the 'first' rule to be executed.

Reply:
Thanks @pcanal for the explanation. If the change would keep the member variable name the same instead (i.e. `float d0_` -> `std::vector<float> d0_`), the rule would be along (right?)
```
source="float d0_"
target="d0_"
```
then the `d0_.clear()` would not be needed?

Reply:
I'm sure the `clear` would still be needed if one were using the 'recommended' way of reading back data which has one reusing the same object over and over. cmsRun doesn't do that and instead forcibly creates a new object for each event.

Reply:
>. If the change would keep the member variable name the same 
> then the d0_.clear() would not be needed?

Actually yes (the in-memory version of `d0_` would also be untouched by the I/0 in that case).

Reply:
Thanks. In the end it is probably cleaner to keep the io rule as it is.